### ISSUE TITLE
[Feature]: Reduce `wandb` folder size by continuously deleting files that have already been synced

### ISSUE THREAD

**Issue by mar-muel** [wandb/wandb#4534]
### Description

Hi team - I'm running into storage space issues on my VM after a while of W&B training. After a few hours my `wandb` local folder is several GB large and eventually leads to the disk filling up and the training run crashing. This happens in particular when logging image snapshots or other storage-intensive assets.

I see that there is a command `wandb sync --clean`. But I'm not sure whether this would possibly delete unsynced data points? Else I guess it could be used in another process with `watch -n 60 "yes | wandb sync --clean"` or similar?

### Suggested Solution

It would be nice if this was already integrated in W&B. At some point, storage will fill up on any machine, which can be a major headache.

### Alternatives

I guess an alternative would be to use something like suggested above?

**Comment by brando90** [comment]
yes, I requested something nearly identical somewhere in the gitissues too.  See my tmp solution here https://github.com/wandb/wandb/issues/4409

**Comment by brando90** [comment]
> Hi team - I'm running into storage space issues on my VM after a while of W&B training. After a few hours my `wandb` local folder is several GB large and eventually leads to the disk filling up and the training run crashing.

me too!

**Comment by andreimargeloiu** [comment]
+1

**Comment by patricksheehan** [comment]
+1, adding that this absolutely tanked a production system that I have (ie filled disk space on a VM rendering it unusable and disrupting forecasts from a production model). I would have 100% expected wandb to automatically clear local files once the run is sync'd, this is a big headache.

FWIW: I've run into so many issues using wandb I'm not sure I'd use it in a future DS application/shop. If the team wants to chat product usability/reliability I'd be happy to donate an hour of time, lmk.

**Comment by thanos-wandb** [comment]
Hi everyone (@mar-muel @brando90 @andreimargeloiu @patricksheehan), apologies that we haven't addressed this issue here yet. This thread somehow wasn't captured by the automation that notify us in our ticketing system. This request in known though to the team as has been asked before by other users. I will add you to the internal ticket so as to reach out to you here once there's any update with an auto-cleanup mechanism.

In the meantime, could you please check if the following manual way would work for you to be executed on your end in a periodic manner? For instance: 
`wandb sync --clean-old-hours N` (to delete runs before N hours. Note that N > duration time of the longest run, just to be make sure you do not delete an active run)

You may also use the argument `--no-include-online` so that you don't clean the online run folders. [More details in our documentation here.](https://docs.wandb.ai/ref/cli/wandb-sync)

Another directory that might grow if you're uploading artifacts is the `.cache` where artifact files are cached to boost performance (instead of actually downloading from the server when used). In order to cleanup this directory you could use the following command ([docs](https://docs.wandb.ai/ref/cli/wandb-artifact/wandb-artifact-cache/wandb-artifact-cache-cleanup)): 
`wandb artifact cache cleanup 10GB` (eg. to delete anything above 10GB, or 0GB to delete all).

I hope this helps for now, but please feel free to ask us here if you had any other questions or issues with these alternative workarounds.

**Comment by ifrit98** [comment]
Where is this request @thanos-wandb?

**Comment by thanos-wandb** [comment]
Hi @ifrit98 there's an internal ticket filed for this feature, but it's not yet selected for development. We will keep you posted here once there are any updates. In the meantime, please let me know if you had any questions or issues with the provided workarounds.

**Comment by gil2rok** [comment]
This is essential! Please implement!

**Comment by shehzi-khan** [comment]
still facing the same issue.
any updates?
appreciate your hard work and response in advanced 👍

**Comment by bysiber** [comment]
The `wandb/` local directory growing unbounded is a real concern for ML engineers running many experiments. It stores run artifacts, logs, and cached data that accumulate quickly:

```bash
# Check wandb directory sizes
find ~ -maxdepth 4 -name "wandb" -type d -exec du -sh {} + 2>/dev/null | sort -rh

# Inside a project:
du -sh wandb/
du -sh wandb/run-*/   # Each run creates a separate directory
du -sh wandb/latest-run/files/

# Clean old runs (keep synced data in cloud)
wandb sync --clean
```

For ML teams running hundreds of experiments, the local wandb directory can easily reach 10-50GB per project. Combined with large model checkpoints, datasets, and Python package caches (especially PyTorch/TensorFlow wheels), ML development machines are especially prone to running out of disk space.

For anyone who needs a quick audit of all these competing cache/data directories, [ClearDisk](https://github.com/bysiber/cleardisk) is a free open-source macOS menu bar app that scans developer caches and shows the full breakdown. Helpful for ML engineers who have Python virtual envs, pip caches, Jupyter data, and wandb artifacts all competing for SSD space.
